src/commands/identify.ts

```diff
@@ -0,0 +1,49 @@
+import { Command, flags } from "@oclif/command"
+import Gravity from "../lib/gravity"
+
+export default class Identify extends Command {
+  static description = "Identify a Gravity resource by its BSON ID"
+
+  static flags = {
+    help: flags.help({ char: "h" }),
+  }
+
+  static args = [{ name: "id" }]
+
+  static collectionsToCheck: CollectionMapping[] = [
+    { name: "Artist", endpoint: "artist" },
+    { name: "Artwork", endpoint: "artwork" },
+    { name: "Partner", endpoint: "partner" },
+  ]
+
+  async run() {
+    const gravity = new Gravity()
+    const { args } = this.parse(Identify)
+    const { id } = args
+
+    const gravityPromises = Identify.collectionsToCheck.map(collection => {
+      const resource = `${collection.endpoint}/${id}`
+      return gravity.get(resource)
+    })
+
+    const gravityResponses = await Promise.all(gravityPromises)
+    const foundIndex = gravityResponses.findIndex(r => r.status === 200)
```

Comment:
The strategy here is to build up all the requests for the various collections, fire em off all at once, and look for one that succeeded.